example-usage/workspace/.devcontainer/devcontainer.json

```
@@ -0,0 +1,31 @@
+{
+	"name": "devcontainer CLI Demo",
+	"build": {
+		"dockerfile": "Dockerfile"
+	},
+
+	// 👇 Dev Container Features - https://containers.dev/implementors/features/
+	"features": {
+		"ghcr.io/devcontainers/features/docker-in-docker": "latest",
```

One thing @alexdima, @chrmarti, @edgonmsft, and I talked about a couple weeks ago was updating all of our examples and templates to pin features to the latest major version so that people who configure their repos based on these will pick up small updates automatically without picking up major breaking changes. Can we update these samples to use the `:1` version which is the current major version of all of the features?

A separate nit is whether we use the shorthand string value or the more explicit object syntax. I'd prefer us to phase out the shorthand form in our examples/templates and not treat the `version` option specially. I think that will help users understand that `version` is just one of the options supported by that feature, and there could be other options or no options depending on which feature they are using.

```suggestion
		"ghcr.io/devcontainers/features/docker-in-docker:1": {
		    "version": "latest"
		},
```

---

We should talk about the phasing out of the shorthand.  When you are hand editing, it's really handy.  Can do the other one though.

---

Yeah we aren't removing the shorthand. The main thing is that too much magic in the examples/docs doesn't help users understand what's going on, so being a bit more explicit/verbose in the examples can be helpful.

---

I'll switch the "latest" ones to `{}` and then add an option to the node one to help clarify. Otherwise just having "version" for each of them ends up looking pretty verbose. I can leave that for the "go" one for comparison though.

---

Done!